dependencies.yaml

```diff
@@ -62,22 +62,20 @@ dependencies:
     common:
       - output_types: [conda, requirements, pyproject]
         packages:
-          - &cmake_ver cmake>=3.26.4
           - cython>=3.0.0
-          - ninja
-          - scikit-build>=0.13.1
-          - tomli
       - output_types: conda
         packages:
           - c-compiler
+          - &cmake_ver cmake>=3.26.4
```

Comment:
Why was `cmake` removed from `[requirements, pyproject]`? We still want/need a way to pin/bound our CMake version even if `scikit-build-core` carries a transitive dependency on `cmake`.

Reply:
It looks like `cmake` is only included in `scikit-build-core[wheels]`.

https://github.com/scikit-build/scikit-build-core/blob/565bd6b67f730acc30f982059b507c04e98fa40f/pyproject.toml#L82

Reply:
scikit-build-core will conditionally install cmake as needed into the build environment. See https://scikit-build-core.readthedocs.io/en/latest/getting_started.html#python-package-configuration

Reply:
Hmm... **(a)** I don't trust this to be correct (we regularly require specific CMake versions, or have to exclude specific versions with known bugs), and **(b)** I just don't _like_ this behavior because it makes too much of the build system "magic." I would prefer to continue specifying `cmake` pinnings (though I don't care either way about ninja).

Reply:
I agree that we need the ability to control the version resolution logic of cmake. If that can be done via toml file, that works for me

Reply:
Sounds good, I can put that back then. We don't have to care about the cases where scikit-build-core is worried about wheels not being available anyway so that's OK.